pkg/migration/migration.go

```diff
@@ -108,6 +115,13 @@ func (m *Service) adjustOwnerReference(om *metav1.ObjectMeta, uidMap map[string]
 	}
 }
 
+func (m *Service) IsMigrationRequired() bool {
+	if _, err := m.appInterface.Deployments(m.releaseNamespace).Get(m.apiserverName, metav1.GetOptions{}); err != nil {
+		return false
+	}
+	return true
+}
```

Comment:
imo the error type should be check to be sure error is about "not found" issue or other problems

```
import "k8s.io/apimachinery/pkg/api/errors"
(...)
if errors.IsNotFound(err) {
	return false, nil
}
if err != nil {
	return false, fmt.Errorf("other type of error: %s", err)
}
```